cylc/flow/network/resolvers.py

```diff
@@ -187,7 +192,7 @@ def get_data_elements(flow, nat_ids, element_type):
     ]
 
 
-class BaseResolvers:
+class BaseResolvers:  # noqa: SIM119 (no real gain + mutable default)
```

Comment:
flake8-simplify is keen on dataclasses.

I.E. this:

```python
from dataclasses import dataclass

@dataclass
class Foo:
    a: str
    b: str
```

Rather than this:

```python
class Foo:

    def __init__(self, a: str, b: str):
        self.a = a
        self.b = b
```

Dataclasses have additional benefits because they add other methods like `__str__` and `__eq__` which saves a lot of boilerplate.

Dataclasses are generally a good idea, however, a lot of our classes (like this one) don't fit the pattern so well.

Nice to be reminded that we could use a dataclass, just stick a `noqa` in if not appropriate.